### PR TITLE
OGL: Clamp rectangle at window dimensions when saving screenshot with crop enabled, use Flag for setting screenshot event without pausing CPU

### PR DESCRIPTION
**Pull request by stenzek**
First commit is changing the GL behavior to match as in this commit: https://github.com/dolphin-emu/dolphin/commit/5dcd3cd4fd30cbe9cff872f952cea9f6d0b91376

Without clamping, glReadPixels was passed negative/out-of-range coordinates, and on nvidia, this just seemed to leave the extra pixels uninitialized, and the GL spec says it only reads 0<=x<width.

Second commit changes the screenshot flag to use a Flag instead of a volatile variable (videosw uses an atomic bool as well), and removes CPU pause/resume calls (it's simply setting a flag, which is checked by the GPU thread). This was causing CPU code to be executed on the UI thread, which at least in debug builds was causing warning messageboxes to be displayed.

[Review on Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/3354)

### CODE REVIEW

File: Source/Core/VideoCommon/RenderBase.cpp [hunk 1]
```diff
@@ -53,13 +53,12 @@ static int OSDTime;
 
 Renderer *g_renderer = nullptr;
 
+std::atomic<bool> Renderer::s_bScreenshot;
```
Comment (delroth) [inline Source/Core/VideoCommon/RenderBase.cpp]: Please use Common::Flag which is one of our abstractions around atomics (Common/Flag.h).

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Comment by bb010g** [comment]
ping

**Comment by RisingFog** [comment]
What's the status of this PR?

**Pull request closed without merging**